### PR TITLE
Solver per branch

### PR DESCRIPTION
**Pull request by norhh**
It seems to be quite costly(memory wise) to maintain solver per branch.If it is possible to make it more optimized or I did some sort of memory leak here then it should probably be great :smiley: .

### CODE REVIEW

**Comment by norhh** [comment]
looks like tests were killed due to the lack of memory of circleci.

File: mythril/analysis/solver.py [hunk 1]
```diff
@@ -2,15 +2,13 @@
 from mythril.exceptions import UnsatError
 import logging
 
+
 def get_model(constraints):
-    s = Solver()
-    s.set("timeout", 100000)
 
-    for constraint in constraints:
-        s.add(constraint)
```
Comment (JoranHonig) [inline mythril/analysis/solver.py]: Now constraints aren't added anymore

Comment (norhh) [reply mythril/analysis/solver.py]: The constraint class automatically does that, so we may not need it.

File: mythril/laser/ethereum/state.py [hunk 2]
```diff
@@ -129,6 +129,37 @@ def as_dict(self):
                     calldata_type=self.calldata_type)
 
 
+class Constraints(list):
```
Comment (JoranHonig) [inline mythril/laser/ethereum/state.py]: Docstrings? :smiley:

File: mythril/laser/ethereum/state.py [hunk 3]
```diff
@@ -183,14 +214,14 @@ class MachineState:
     """
     MachineState represents current machine state also referenced to as \mu
     """
-    def __init__(self, gas):
+    def __init__(self, gas, pc=0, stack=[], memory=[], constraints=None, depth=0):
```
Comment (JoranHonig) [inline mythril/laser/ethereum/state.py]: There are mutable defaults here. You could use: memory = None
and later
self.memory = memory or []

Comment (norhh) [reply mythril/laser/ethereum/state.py]: oops, this I forgot, this PR was older than that one :sweat_smile:

Comment (JoranHonig) [reply mythril/laser/ethereum/state.py]: bump

Comment (JoranHonig) [reply mythril/laser/ethereum/state.py]: @norhh This is still mutable

**Comment by norhh** [comment]
oops this PR was for master, will add a new PR to develop

**Pull request closed without merging**